packages/commonwealth/server/workers/sitemap/createDatabasePaginator.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  models,
+  ProfileAttributes,
+  ThreadInstance,
+} from '@hicommonwealth/model';
+import { QueryTypes } from 'sequelize';
+import { getThreadUrl } from 'utils';
+
+export interface Link {
+  readonly id: number;
+  readonly url: string;
+  readonly updated_at: string;
+}
+
+/**
+ * A single page which holds a batch of links that can be converted to a sitemap.
+ */
+export interface Page {
+  readonly links: ReadonlyArray<Link>;
+}
+
+/**
+ * Interface representing a Paginator.
+ *
+ * A Paginator is used to navigate through a collection of pages.  This way
+ * we can avoid fetching all records into memory.
+ */
+export interface Paginator {
+  readonly hasNext: () => Promise<boolean>;
+  readonly next: () => Promise<Page>;
+}
+
+interface TableAdapter {
+  toRecord: (obj: object) => Link;
+  executeQuery: (ptr: number, limit: number) => Promise<ReadonlyArray<object>>;
+}
+
+export function createDatabasePaginatorDefault(limit: number = 50000) {
+  const threads = createDatabasePaginatorWithAdapter(
+    createThreadsTableAdapter(),
+    limit,
+  );
+  const profiles = createDatabasePaginatorWithAdapter(
+    createProfilesTableAdapter(),
+    limit,
+  );
+
+  return { threads, profiles };
+}
+
+function createThreadsTableAdapter(): TableAdapter {
+  type ThreadInstancePartial = Pick<
+    ThreadInstance,
+    'id' | 'updated_at' | 'title' | 'community_id'
+  >;
+
+  function toRecord(obj: object) {
+    const thread = obj as ThreadInstancePartial;
+    const url = getThreadUrl({
+      chain: thread.community_id,
+      id: thread.id,
+      title: thread.title,
+    });
+    return {
+      id: thread.id,
+      url,
+      updated_at: thread.updated_at.toISOString(),
+    };
+  }
+
+  async function executeQuery(
+    ptr: number,
+    limit: number,
+  ): Promise<ReadonlyArray<object>> {
+    return await models.sequelize.query(
+      `
```

Comment:
This raw SQL query is simple enough that we should implement it directly in Sequelize:
```
models.Thread.findAll({
  attributes: ['id', 'updated_at', 'title', 'community_id'],
  where: {
    id: { [Op.gt]: ptr }
  },
  order: [['id', 'ASC']],
  limit,
});
```

I haven't tested this query but this should be approximately what it looks like.

Reply:
True... I changed it so I'll try to get it working with findAll